ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -115,8 +115,30 @@ function filterGraph(app: models.Application, filteredIndicatorParent: string, g
     }
 }
 
-function compareNodes(first: ResourceTreeNode, second: ResourceTreeNode) {
-    return `${(first.orphaned && '1') || '0'}/${nodeKey(first)}`.localeCompare(`${(second.orphaned && '1') || '0'}/${nodeKey(second)}`);
+export function compareNodes(first: ResourceTreeNode, second: ResourceTreeNode) {
+    function orphanedToInt(orphaned?: boolean) {
+        return (orphaned && 1) || 0;
+    }
+    function compareRevision(a: string, b: string) {
+        const numberA = Number(a);
+        const numberB = Number(b);
+        if (isNaN(numberA) || isNaN(numberB)) {
+            return a.localeCompare(b);
+        }
+        return Math.sign(numberA - numberB);
+    }
+    function getRevision(a: ResourceTreeNode) {
+        const value = a.info.filter(b=>b.name=="Revision" && b)[0]?.value;
```

Comment:
I'm not a typescript expert... And this is fairly magical. I think it's right. It could be totally wrong, it's definitely a bit ugly. Happy to reformat.

Reply:
This requires ts3.7, argo is currently using 3.5. We can upgrade argo first in general and revisit that as a whole.